elephant/change_point_detection.py

```diff
@@ -0,0 +1,514 @@
+        test_param : np.array 3 * num of window,
+            first row: list of `h`, second and third rows: empirical means and
+            variances of the limit process correspodning to `h`. This will be 
+            used to normalize the `filter_process` in order to give to the every
+            maximum the same impact on the global statistic.
+    """
+
+    u = 1 * pq.s
+    try:
+        window_sizes_sec = window_sizes.rescale(u)
+    except ValueError:
+        raise ValueError("H must be a list of times")
+    window_sizes_mag = window_sizes_sec.magnitude
+    try:
+        t_final_sec = t_final.rescale(u)
+    except ValueError:
+        raise ValueError("T must be a time quantity")
+    t_final_mag = t_final_sec.magnitude
+
+    if t_final_mag <= 0:
+        raise ValueError("T needs to be stricktly poisitive")
+    if alpha * (100 - alpha) < 0:
+        raise ValueError("alpha needs to be in (0,100)")
+    if not isinstance(n_surrogates, int):
+        raise TypeError("n_surrogates needs to be an integer")
+    if np.min(window_sizes_mag) <= 0:
+        raise ValueError("window's size needs to be stricktly poisitive")
+    if np.max(window_sizes_mag) >= t_final / 2:
+        raise ValueError("window's size tooo large")
+    if dt is not None:
+        try:
+            dt = dt.rescale(u)
+        except ValueError:
+            raise ValueError("dt must be a time quantity")
+        for h in window_sizes_mag:
+            if (h / dt).magnitude - int(h / dt) != 0:
+                raise ValueError(
+                    "Every window size h must be a multiple of dt")
+
+    # Generate a matrix: n X m where n = n_surrogates is the number of
+    # simulated limit processes and m is the number of choosen window size.
+    # Entrances are: M*(n,h) = max(t in T)[`limit_process_h`(t)],
+    # for each h in H and surrogate n
+    maxima_matrix = []
+
+    for i in range(n_surrogates):
+            mh_star = []
+            simu = _limit_processes(window_sizes, t_final, dt)
+            for i, h in enumerate(window_sizes_mag):
+                # max over time of the limit process generated with window h
+                m_h = np.max(simu[i])
+                mh_star.append(m_h)
+            maxima_matrix .append(mh_star)
+
+    maxima_matrix = np.asanyarray(maxima_matrix)
+    matrix = maxima_matrix.T
+    # matrix normalization by mean and variance of the limit process, in order
+    # to give, for every h, the same impact on the global maximum
+    matrix_normalized = []
+    # these parameters will be used to normalize both the limit_processes (H0)
+    # and the filter_processes
+    null_mean = []
+    null_var = []
+
+    for i, h in enumerate(window_sizes_mag):
+        mean = np.mean(matrix, axis=1)[i]
+        var = np.var(matrix, axis=1)[i]
+        matrix_normalized.append((matrix[i] - mean) / np.sqrt(var))
+        null_mean.append(mean)
+        null_var.append(var)
+    matrix_normalized = np.asanyarray(matrix_normalized)
```

Comment:
This part of code is unnecessarily complicated and redundant.
The same result can be obtained by
```
null_mean = maxima_matrix.mean(axis=0)
null_var = maxima_matrix.var(axis=0)
matrix_normalized = (maxima_matrix - null_mean) / np.sqrt(null_var)
```